CE-657 - Move Application leader election tutorial to docs (#21366)
hashicorp/consul: website/content/docs/dynamic-app-config/sessions/index.mdx

> The primitives provided by sessions and the locking mechanisms of the KV
> store can be used to build client-side leader election algorithms.

Reviewer: ```suggestion
You can use the primitives provided by sessions and the locking mechanisms of the KV
store to build client-side leader election algorithms.
```
